### PR TITLE
Extract avro validator 

### PR DESCRIPTION
**Pull request by attilakreiner**
## Description

Extract the avro validator from `runtime/engine` to `incubator/validator-avro`.

Fixes # (issue)

### CODE REVIEW

File: runtime/binding-kafka/src/test/java/io/aklivity/zilla/runtime/binding/kafka/internal/stream/CacheFetchIT.java [hunk 1]
```diff
@@ -315,24 +315,24 @@ public void shouldReceiveMessageValueStringInvalid() throws Exception
...
     }
 
     @Test
-    @Configuration("cache.options.type.avro.yaml")
+    @Configuration("cache.options.type.test.yaml")
     @Specification({
-        "${app}/message.value.avro.invalid/client",
-        "${app}/message.value.avro.invalid/server"})
+        "${app}/message.value.test.invalid/client",
+        "${app}/message.value.test.invalid/server"})
```
Comment (jfallows) [inline runtime/binding-kafka/src/test/java/io/aklivity/zilla/runtime/binding/kafka/internal/stream/CacheFetchIT.java:333]: Please rename scenario to `message.value.invalid`.

File: runtime/binding-kafka/src/test/java/io/aklivity/zilla/runtime/binding/kafka/internal/stream/CacheFetchIT.java [hunk 2]
```diff
@@ -315,24 +315,24 @@ public void shouldReceiveMessageValueStringInvalid() throws Exception
     }
 
     @Test
-    @Configuration("cache.options.type.avro.yaml")
+    @Configuration("cache.options.type.test.yaml")
     @Specification({
-        "${app}/message.value.avro/client",
-        "${app}/message.value.avro/server"})
+        "${app}/message.value.test/client",
+        "${app}/message.value.test/server"})
```
Comment (jfallows) [inline runtime/binding-kafka/src/test/java/io/aklivity/zilla/runtime/binding/kafka/internal/stream/CacheFetchIT.java:321]: Please rename scenario to `message.value.valid`.

File: runtime/engine/pom.xml [hunk 3]
```diff
@@ -27,7 +27,7 @@
     <maven.compiler.source>11</maven.compiler.source>
     <maven.compiler.target>11</maven.compiler.target>
     <jacoco.coverage.ratio>0.70</jacoco.coverage.ratio>
-    <jacoco.missed.count>4</jacoco.missed.count>
+    <jacoco.missed.count>8</jacoco.missed.count>
```
Comment (jfallows) [inline runtime/engine/pom.xml:30]: Please review this after moving `ValidatorFactory.instantiate()` to `engine`.

File: specs/binding-kafka.spec/src/test/java/io/aklivity/zilla/specs/binding/kafka/config/SchemaTest.java [hunk 4]
```diff
@@ -117,7 +117,7 @@ public void shouldValidateClientWhenTopic()
     @Test
     public void shouldValidateCacheOptionsCatalog()
     {
-        JsonObject config = schema.validate("cache.options.type.avro.yaml");
+        JsonObject config = schema.validate("cache.options.type.test.yaml");
```
Comment (jfallows) [inline specs/binding-kafka.spec/src/test/java/io/aklivity/zilla/specs/binding/kafka/config/SchemaTest.java:120]: Please rename config to `cache.options.validate.yaml`.

**Review by jfallows — CHANGES_REQUESTED** [review]

File: specs/binding-kafka.spec/src/test/java/io/aklivity/zilla/specs/binding/kafka/config/SchemaTest.java (hunk 4 above)
Comment (attilakreiner) [reply specs/binding-kafka.spec/src/test/java/io/aklivity/zilla/specs/binding/kafka/config/SchemaTest.java]: done

File: runtime/engine/pom.xml (hunk 3 above)
Comment (attilakreiner) [reply runtime/engine/pom.xml]: done

File: runtime/engine/src/main/java/io/aklivity/zilla/runtime/engine/Engine.java [hunk 5]
```diff
@@ -180,6 +184,8 @@ public final class Engine implements Collector, AutoCloseable
         schemaTypes.addAll(metricGroups.stream().map(MetricGroup::type).filter(Objects::nonNull).collect(toList()));
         schemaTypes.addAll(vaults.stream().map(Vault::type).filter(Objects::nonNull).collect(toList()));
         schemaTypes.addAll(catalogs.stream().map(Catalog::type).filter(Objects::nonNull).collect(toList()));
+        schemaTypes.addAll(validatorFactory.validatorSpis().stream().map(ValidatorFactorySpi::schema)
+            .filter(Objects::nonNull).collect(toList()));
```
Comment (jfallows) [inline runtime/engine/src/main/java/io/aklivity/zilla/runtime/engine/Engine.java:188]: Looks good, let's make sure to remove the filter when we have core extracted so every validator factory comes with a schema patch.

File: runtime/engine/src/main/java/io/aklivity/zilla/runtime/engine/validator/ValidatorFactorySpi.java [hunk 6]
```diff
@@ -25,6 +26,11 @@ public interface ValidatorFactorySpi
 {
     String type();
 
+    default URL schema()
+    {
+        return null;
+    }
+
```
Comment (jfallows) [inline runtime/engine/src/main/java/io/aklivity/zilla/runtime/engine/validator/ValidatorFactorySpi.java:33]: Also to be removed when core validators are extracted.

**Review by jfallows — APPROVED** [review]

**Pull request merged**